### ISSUE TITLE
Migrate to module-supported Nullable annotation package

### ISSUE THREAD

**Issue by jhy** [jhy/jsoup#2028]
To mitigate errors/warnings in IntelliJ (but not present in the real Maven build), I'm temporarily adding `requires static jsr305;` to `module-info.java`, so that the nullable annotations are visible. But, we don't want to produce a published library with this as it's an automodule.

Need to select another Nullable package (#1992 jspecify?) before next release. Something that IDEs and Kotlin and other various tools will recognize.

**Comment by jhy** [comment]
In testing this, to get warnings to show up in the current IntelliJ version for jspecify 0.3.0 (the latest release), I needed to explicitly add them to IntelliJ:

Settings -> Editor -> Inspections -> Probable bugs -> Nullability Problems -> \@NotNull/\@Nullable problems:
Configure Annotations
Added `org.jspecify.annotations.Nullable` to the Nullable tab. And `org.jspecify.annotations.NullMarked` and `org.jspecify.annotations.NonNull` to the Not Null tab.

Pity that jspecify isn't directly supported currently in IntelliJ. [This bug](https://youtrack.jetbrains.com/issue/IDEA-323691) says it is implemented in EAP 233. Current production build is 232.
